README
---
*Creating HPI packages

Driver HPI found at
http://www.audioscience.com/internet/download/linux_drivers.htm

downloaded the source from
https://launchpad.net/~audioscience/+archive/ubuntu/release/+packages

and re-compiled that for stretch

dh_clean
update xenial to stretch in debain/changelog
debuild -b -uc -us
sudo dpkg -i libhpi libhpi-dev

---
*Installing pre-requisites


sudo apt-get install devscripts libautotool-bin dpatch libatg1-dev



sudo apt-get install xsltproc docbook-xsl fop

sudo apt-get install libqt4dev libjack-jackd2-dev libtwolame-dev libmp3lame-dev libxt-dev libvorbis-dev libsamplerate-dev libsndfile1-dev libcdparanoia-dev libid3-3.8.3-dev libcurl4-nss-dev libsoundtouch-dev libasound2-dev libmad0-dev libmp4v2-dev libflac++-dev libfaad-dev libhpi libhpi-dev libpam0g-dev libssl-dev libhpi libhpi-dev

---
*Testing the build

export DOCBOOK_STYLESHEETS=/usr/share/xml/docbook/stylesheet/docbook-xsl/

./autogen.sh
autoconf
./configure --with-Qt-include-dir="/usr/include/qt4/" --with-Qt-lib-dir=/usr/lib/x86_64-linux-gnu/ --with-Qt-lib="QtCore -lQtGui -lQtNetwork -lQtSql -lQt3Support" CXXFLAGS="-I/usr/include/qt4/Qt -I/usr/include/qt4/QtSql -I/usr/include/qt4/QtGui -I/usr/include/qt4/Qt3Support -DQT3_SUPPORT=1"

---
*Do it for real

make clean
make maintainer-clean
dh_clean

debuild -S -uc -us

export DOCBOOK_STYLESHEETS=/usr/share/xml/docbook/stylesheet/docbook-xsl/
cd doc/dockbook
make 
cd ../..

debuild -b -uc -us

sudo dpkg -i rivendell_2.16.0int06-1_amd64.deb librivendell_2.16.0int06-1_amd64.deb rivendell-server_2.16.0int06-1_amd64.deb

--
*Setting MariaDB 

sudo nano /etc/mysql/mariadb.conf.d/50-server.cnf 
< character-set-server  = utf8mb4
< collation-server      = utf8mb4_general_ci

> character-set-server  = utf8
> collation-server      = utf8_general_ci


---
*Installing Rivendell

https://tecwhisperer.com/install-rivendell-automation-part-1-system-preparation/

---
Note: if daemons do not start

sudo mkdir /var/run/rivendell
sudo chmod a+w /var/run/rivendell

---



This is the README file for the Rivendell package.

This software comes with ABSOLUTELY NO WARRANTY.  See the file COPYING for
details.

Welcome to this release of the Rivendell Project.  Rivendell aims
to be a robust, functionally complete digital audio system for use in 
professional radio broadcast environments.

See the file INSTALL for installation instructions.  If you are upgrading from
a 1.x version of Rivendell, also see the file 'UPGRADING' for important notes
and pointers.


OVERVIEW
Rivendell contains a full set of functionality needed to operate a radio
automation system, consisting of the following components:

RDAdmin
A comprehensive application for the administration and configuration
of Rivendell.

RDLibrary
The production audio interface.

RDCatch
The automatic recorder and task scheduler interface.

RDLogin
A small utility for logging users into and out of the system.

RDLogEdit
A air log creation, editing and voicetracker tool.

RDLogManager
A utility for the automatic generation of logs, based on templates.
It also supports importation of schedule information from a wide
variety of third-party traffic and music scheduling systems.

RDAirPlay
The on-air playout application.

RDPanel
A 'fullscreen' SoundPanel cartwall application.

RDCastManager
A utility for posting and managing podcast episodes.

RDGpiMon
A utility for monitoring GPI line status.

RDHPIInfo
A utility for configuring AudioScience audio adapters.

RDAlsaConfig
A utility for configuring ALSA audio adapters.

RDImport
A command-line utility for batch importing audio along with
accompanying metadata.  Do:

	rdimport --help

from a command prompt for a list of options.

RMLSend
A GUI and command-line utility for sending Rivendell Macro Language
(RML) commands.

RDSoftKeys
A lightweight button panel utility for sending RML commands.


debian/changelog
rivendell (2.16.0int08-1) unstable; urgency=medium

  * Fix some problems in porting to Qt4

 -- Nicola Bassan <info@radioincorso.it>  Sat, 29 Jul 2017 21:00:00 +0100

rivendell (2.16.0int06-1) unstable; urgency=medium

  * New upstream release

 -- Nicola Bassan <info@radioincorso.it>  Mon, 10 Jul 2017 21:00:00 +0100

rivendell (2.10.3-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Thu, 26 Feb 2015 23:46:41 +0100

rivendell (2.10.2-2) unstable; urgency=medium

  * Include upstream fix for column HAVE_CODE2 in DB creation (commit
    7d46f597a25e2)

 -- Alban Peignier <alban@tryphon.eu>  Wed, 10 Dec 2014 16:51:37 +0100

rivendell (2.10.2-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Wed, 26 Nov 2014 17:43:21 +0100

rivendell (2.10.1-2) unstable; urgency=medium

  * Include patch for Pull Request 42 - Improve log save performance issue (fixes #1248).

 -- Alban Peignier <alban@tryphon.eu>  Sat, 15 Nov 2014 20:30:59 +0100

rivendell (2.10.1-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Wed, 05 Nov 2014 00:49:21 +0100

rivendell (2.10.0-2) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Sun, 02 Nov 2014 10:55:29 +0100

rivendell (2.10.0-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Sat, 01 Nov 2014 15:04:04 +0100

rivendell (2.9.3-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Fri, 19 Sep 2014 10:15:04 +0200

rivendell (2.9.2-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Fri, 12 Sep 2014 09:53:03 +0200

rivendell (2.9.0-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Sat, 07 Jun 2014 10:10:28 +0200

rivendell (2.8.2-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Sat, 17 May 2014 10:19:29 +0200

rivendell (2.8.1-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Fri, 28 Feb 2014 18:59:48 +0100

rivendell (2.8.0-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Fri, 14 Feb 2014 23:47:05 +0100

rivendell (2.7.0-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Mon, 20 Jan 2014 22:37:26 +0100

rivendell (2.6.2-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Sat, 04 Jan 2014 10:11:09 +0100

rivendell (2.6.1-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Fri, 03 Jan 2014 12:22:04 +0100

rivendell (2.6.0-1) unstable; urgency=medium

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Fri, 13 Dec 2013 18:53:49 +0100

rivendell (2.5.5-2) unstable; urgency=low

  * Debian/Ubuntu rebuild

 -- Alban Peignier <alban@tryphon.eu>  Thu, 28 Nov 2013 08:11:38 +0100

rivendell (2.5.5-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Wed, 20 Nov 2013 18:39:18 +0100

rivendell (2.5.3-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Mon, 07 Oct 2013 08:45:04 +0200

rivendell (2.5.1-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Wed, 22 May 2013 19:59:44 +0200

rivendell (2.5.0-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Tue, 02 Apr 2013 09:41:45 +0200

rivendell (2.4.0-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Sun, 24 Mar 2013 09:45:13 +0100

rivendell (2.2.1-1) unstable; urgency=low

  * Provides RLM in rivendell packages (fixes #616)
  * New upstream release (fixes #617)
  * Fixes prefixes in script and select Makefiles

 -- Alban Peignier <alban@tryphon.eu>  Fri, 26 Oct 2012 12:40:33 +0200

rivendell (2.2.0-1) unstable; urgency=low

  * New upstream release
  * Add libqt3-headers in suggested packages. Fixes #414
  * Add explicit librivendell dependency with binary:Version. Fixes #415

 -- Alban Peignier <alban@tryphon.eu>  Mon, 08 Oct 2012 17:12:03 +0200

rivendell (2.1.4-2) unstable; urgency=low

  * Fixes rivendell-doc package contents (fixes #4)
  * Use build QTDIR as default value in RDHotKeyList on runtime (fixes #1)

 -- Alban Peignier <alban@tryphon.eu>  Sat, 23 Jun 2012 11:44:15 +0200

rivendell (2.1.2-2) unstable; urgency=low

  * Add experimental french translation

 -- Alban Peignier <alban@tryphon.eu>  Mon, 27 Feb 2012 21:29:24 +0100

rivendell (2.1.2-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Sat, 31 Dec 2011 11:50:16 +0100

rivendell (2.1.1-2) unstable; urgency=low

  * Include icons and menu entries

 -- Alban Peignier <alban@tryphon.eu>  Fri, 11 Nov 2011 13:50:16 +0100

rivendell (2.1.1-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Thu, 03 Nov 2011 21:11:00 +0100

rivendell (2.1.0-1) unstable; urgency=low

  * New upstream release

 -- Alban Peignier <alban@tryphon.eu>  Wed, 02 Nov 2011 20:48:05 +0100

rivendell (2.0.2-2) unstable; urgency=low

  * Add postrm script for rivendell and rivendell-server
  * Include libtwolame in dependencies
  * Recommends libraries for MPEG support
  * Add rivendell-daemons tool

 -- Alban Peignier <alban@tryphon.eu>  Sun, 30 Oct 2011 12:00:51 +0100

rivendell (2.0.2-1) unstable; urgency=low

  * Rewrite debian packaging

 -- Alban Peignier <alban@tryphon.eu>  Sat, 27 Aug 2011 14:45:14 +0200

rivendell (1.7.2-3) unstable; urgency=low

  * Remove alsasound from rivendell.init, uses --quiet start-stop-daemon option (Closes #2)
  * Remove libtool version test (Closes #5)
  * Include rlm files in librivendell (Closes #1)

 -- Alban Peignier <alban@tryphon.eu>  Sun, 23 Jan 2011 19:34:11 +0100


lib/rd.h
// rd.h
//
// System-Wide Values for Rivendell
//
//   (C) Copyright 2002-2016 Fred Gleason <fredg@paravelsystems.com>
//
//   This program is free software; you can redistribute it and/or modify
//   it under the terms of the GNU General Public License version 2 as
//   published by the Free Software Foundation.
//
//   This program is distributed in the hope that it will be useful,
//   but WITHOUT ANY WARRANTY; without even the implied warranty of
//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//   GNU General Public License for more details.
//
//   You should have received a copy of the GNU General Public
//   License along with this program; if not, write to the Free Software
//   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
//

#ifndef RD_H
#define RD_H

/*
 * Default Configuration File
 */
#define RD_CONF_FILE "/etc/rd.conf"
#define RD_WIN_CONF_FILE "rd.ini"

/*
 * Default ALSA asoundrc file
 */
#define RD_ASOUNDRC_FILE "/etc/asound.conf"

/*
 * PID File Locations
 */
#define RD_PID_DIR "/var/run/rivendell"
#define RD_CAED_PID "/var/run/rivendell/caed.pid"
#define RD_RIPCD_PID "/var/run/rivendell/ripcd.pid"
#define RD_RDCATCHD_PID "/var/run/rivendell/rdcatchd.pid"

/*
 * Where audio gets stored
 */
#define RD_AUDIO_ROOT "/var/snd"

/*
 * Audio File Extension
 */
#define RD_AUDIO_EXTENSION "wav"

/*
 * Allow Nonstandard Bitrates?
 */
#define RD_ALLOW_NONSTANDARD_RATES false

/*
 * Default mySQL Settings
 */
#define DEFAULT_MYSQL_HOSTNAME "localhost"
#define DEFAULT_MYSQL_DATABASE "Rivendell"
#define DEFAULT_MYSQL_USERNAME "rduser"
#define DEFAULT_MYSQL_PASSWORD "letmein"
#define DEFAULT_MYSQL_DRIVER "QMYSQL3"
#define DEFAULT_MYSQL_HEARTBEAT_INTERVAL 360
#define MYSQL_BUILTIN_DATABASE "mysql"
#define POSTGRESQL_BUILTIN_DATABASE "template1"

/*
 * Default Logging Settings
 */
#define DEFAULT_LOG_FACILITY "syslog"
#define DEFAULT_LOG_CORE_DUMP_DIRECTORY "/"
#define DEFAULT_LOG_PATTERN "%n-%Y%M%d.log"

/*
 * ALSA Settings
 */
#define RD_ALSA_DEFAULT_PERIOD_QUANTITY 4
#define RD_ALSA_DEFAULT_PERIOD_SIZE 1024
#define RD_ALSA_FADE_INTERVAL 100
#define RD_ALSA_SAMPLE_RATE_TOLERANCE 100

/*
 * Date Limits
 */
#define RD_MAX_YEAR 8000

/*
 * JACK Settings
 */
#define RD_JACK_FADE_INTERVAL 100

/*
 * RIPCD TCP Port
 */
#define RIPCD_TCP_PORT 5006

/*
 * CAED TCP Port
 */
#define CAED_TCP_PORT 5005

/*
 * RdCatchd TCP Port
 */
#define RDCATCHD_TCP_PORT 6006

/*
 * Minimum event ID for dynamic (RML-controlled) recordings
 */
#define RDCATCHD_DYNAMIC_BASE_ID 1000000000

/*
 * Default Local GPIO Device
 */
#define RD_DEFAULT_GPIO_DEVICE "/dev/gpio0"

/*
 * Max number of possible GPIO lines (GPI OR GPO)
 */
#define MAX_GPIO_PINS 32768

/*
 * Max number of possible audio adapters
 */
#define RD_MAX_CARDS 8

/*
 * Max number of possible audio streams/card/type
 */
#define RD_MAX_STREAMS 16

/*
 * Max number of possible audio ports/card/type
 */
#define RD_MAX_PORTS 8

/*
 * Max number of possible TTYs
 */
#define MAX_TTYS 8

/*
 * Max number of netcatcher decks per workstation
 */
#define MAX_DECKS 8

/*
 * Max number of inputs or outputs in a switcher
 */
#define MAX_ENDPOINTS 2048

/*
 * Max number of attached switcher matrices per workstation
 */
#define MAX_MATRICES 8

/*
 * Max number of macro timers
 */
#define RD_MAX_MACRO_TIMERS 16

/*
 * Reference Analog Level at 0 dBFS (1/100 dB)
 * (thus, -16 dBFS = +4 dBu)
 */
#define RD_BASE_ANALOG 1600 

/*
 * Maximum number of SoundPanels of a given type
 */
#define MAX_PANELS 50

/*
 * Maximum Cart Number
 */
#define RD_MAX_CART_NUMBER 999999

/*
 * Maximum Cut Number
 */
#define RD_MAX_CUT_NUMBER 999

/*
 * In negative 1/100ths of a dB
 */
#define REFERENCE_LEVEL 1600

/*
 * Typomatic Rate Interval, in mS
 */
#define TYPO_RATE_1 300
#define TYPO_RATE_2 100

/*
 * ALSA Values
 */
#define ALSA_PLAY_PCM_DEVICE "rdp"
#define ALSA_RECORD_PCM_DEVICE "rdr"

/*
 * Default Administrative User
 */
#define RDA_LOGIN_NAME "admin"
#define RDA_PASSWORD ""
#define RDA_FULLNAME "Rivendell Administrator"
#define RDA_DESCRIPTION "Default Administrator Account"

/*
 * Default System User
 */
#define RD_USER_LOGIN_NAME "user"
#define RD_USER_PASSWORD ""
#define RD_USER_FULL_NAME "Rivendell User"
#define RD_USER_DESCRIPTION "Default User Account"

/*
 * Default System Identities
 */
#define RD_DEFAULT_AUDIO_OWNER "user"
#define RD_DEFAULT_AUDIO_GROUP "users"
#define RD_DEFAULT_LABEL "Default Configuration"

/*
 * Default Audio Store Settings
 */
#define RD_DEFAULT_AUDIO_STORE_MOUNT_OPTIONS "defaults"

/*
 * Default RDSelect Configuration Directory
 */
#define RD_DEFAULT_RDSELECT_DIR "/etc/rivendell.d"

/*
 * Default Workstation
 */
#define RD_STATION_NAME "DEFAULT"
#define RD_STATION_DESCRIPTION "Default Workstation"

/*
 * Default Program Service
 */
#define RD_SERVICE_NAME "Production"
#define RD_SERVICE_DESCRIPTION "Default Audio Service"

/*
 * RDSelect Label
 * (Used for the RDSelect entry in RDAdmin->ManageHosts)
 */
#define RD_RDSELECT_LABEL "[RDSelect]"

/*
 * Default Maximum POST Length (bytes)
 */
#define RD_DEFAULT_MAX_POST_LENGTH 10000000

/*
 * Pause Time for Starting Daemons (secs)
 */
#define RD_DAEMON_PAUSE_TIME 1

/*
 * Location of the proc filesystem
 */
#define RD_PROC_DIR "/proc"

/*
 * Location for Lock Files
 */
#define RD_LOCKFILE_DIR "/var/lock"

/*
 * Rivendell Macro Language (RML)
 */
#define RD_RML_ECHO_PORT 5858
#define RD_RML_NOECHO_PORT 5859
#define RD_RML_REPLY_PORT 5860
#define RD_RML_MAX_ARGS 100
#define RD_RML_MAX_LENGTH 2048

/*
 * Maximum Permissible Time Offset (+/- mS)
 */
#define RD_MAX_TIME_OFFSET 10000

/*
 * The file to save geometry state in
 */
#define RD_GEOMETRY_FILE ".rivendell"

/*
 * Meter Update Interval (msecs)
 */
#define RD_METER_UPDATE_INTERVAL 20

/*
 * The metering block shared memory key
 */
#define RD_METER_SHM_KEY 0x00005005

/*
 * Maximum length of import log file line
 */
#define RD_MAX_IMPORT_LINE_LENGTH 1024

/*
 * The minimum fader gain at the end of a segue transition
 * (in hundreths of a dB)
 */
#define RD_FADE_DEPTH -3000

/*
 * The mixer 'off' level for full muting
 */
#define RD_MUTE_DEPTH -10000

/*
 * The fade profile (for HPI adapters only)
 */
#define RD_FADE_TYPE RDHPISoundCard::Log

/*
 * This defines the maximum frequency (in Hz) of the broadcast audio
 * channel being fed by Rivendell.  It is used for things like calculating
 * how expensive an algorithm is needed for sample rate conversion.
 *
 * For modern FM facilities running MPX stereo, 15000 Hz is a reasonable
 * value.
 */
#define RD_MAX_BANDPASS 15000

/*
 * The limits on timescaling
 */
#define RD_TIMESCALE_MIN 0.833
#define RD_TIMESCALE_MAX 1.250
#define RD_TIMESCALE_DIVISOR 100000.0

/*
 * Max number of default services
 */
#define RD_MAX_DEFAULT_SERVICES 32

/*
 * Ripper Settings
 */
#define RIPPER_BAR_INTERVAL 500
#define RIPPER_TEMP_WAV "rdlibrary_rip.wav"
#define RIPPER_TEMP_PEAK "rdlibrary_rip.dat"
#define RIPPER_CDDB_USER "rdlibrary"
#define RIPPER_MAX_SECTORS 20

/*
 * CAE Values
 */
#define CAE_MAX_ARGS 10
#define CAE_MAX_LENGTH 256
#define CAE_POLL_INTERVAL 50
#define CAE_MAX_CONNECTIONS 128

/*
 * Default Sample Rate
 */
#define RD_DEFAULT_SAMPLE_RATE 48000

/*
 * Default Channels
 */
#define RD_DEFAULT_CHANNELS 2

/*
 * Marker Colors
 */
#define RD_SEGUE_MARKER_COLOR Qt::cyan
#define RD_TALK_MARKER_COLOR Qt::blue
#define RD_START_END_MARKER_COLOR Qt::red
#define RD_FADE_MARKER_COLOR Qt::yellow
#define RD_HOOK_MARKER_COLOR Qt::magenta

/*
 * Cart Status Colors
 */
#define RD_CART_ERROR_COLOR Qt::red
#define RD_CART_CONDITIONAL_COLOR Qt::yellow
#define RD_CART_FUTURE_COLOR "#00FFFF"
#define RD_CART_EVERGREEN_COLOR "#008000"
#define RD_CART_INVALID_SERVICE_COLOR Qt::magenta

/*
 * RDAirPlay Colors
 */
#define RDPANEL_SETUP_FLASH_COLOR Qt::blue
#define RDPANEL_RESET_FLASH_COLOR Qt::blue
#define RDPANEL_PLAY_BACKGROUND_COLOR Qt::red
#define RDPANEL_PAUSED_BACKGROUND_COLOR Qt::cyan
#define BUTTON_STOPPED_BACKGROUND_COLOR Qt::green
#define BUTTON_PLAY_BACKGROUND_COLOR Qt::red
#define BUTTON_PLAY_TEXT_COLOR Qt::black
#define BUTTON_PAUSE_BACKGROUND_COLOR Qt::cyan
#define BUTTON_PAUSE_TEXT_COLOR Qt::black
#define BUTTON_FROM_BACKGROUND_COLOR Qt::magenta
#define BUTTON_FROM_TEXT_COLOR Qt::black
#define BUTTON_TO_BACKGROUND_COLOR Qt::yellow
#define BUTTON_TO_TEXT_COLOR Qt::black
#define BUTTON_DISABLED_BACKGROUND_COLOR Qt::darkGray
#define BUTTON_DISABLED_TEXT_COLOR Qt::white
#define BUTTON_ERROR_BACKGROUND_COLOR Qt::red
#define BUTTON_ERROR_TEXT_COLOR Qt::black

/*
 * RDAirPlay Log Machines
 */
#define RDAIRPLAY_LOG_QUANTITY 3

/*
 * Cue Editor Colors
 */
#define RD_CUEEDITOR_KNOB_COLOR Qt::blue
#define RD_CUEEDITOR_PLAY_MARKER Qt::black
#define RD_CUEEDITOR_START_MARKER Qt::red
#define RD_CUEEDITOR_BUTTON_FLASH_PERIOD 200
#define RD_CUEEDITOR_AUDITION_PREROLL 5000

/*
 * Log Colors
 */
#define RD_CUSTOM_TRANSITION_COLOR Qt::blue

/*
 * Default Text Editors
 */
#define RD_LINUX_EDITOR "xterm -e vi"
#define RD_WIN32_EDITOR "notepad"

/*
 * System-wide Maintenance Interval (mS)
 */
#define RD_MAINT_MIN_INTERVAL 900000
#define RD_MAINT_MAX_INTERVAL 3600000

/*
 * Audio File Filter for QFileDialog
 */
#define RD_AUDIO_FILE_FILTER "Sound Files (*.mp* *.MP* *.wav *.WAV *.ogg *.OGG *.flac *.FLAC *.atx *.ATX *.tmc *.TMC *.aif* *.AIF* *.m4a *.M4A *.aac *.AAC)\nAIFF Files (*.aif* *.AIF*)\nATX Files (*.atx *.ATX)\nMPEG Files (*.mp* *.MP*)\nOggVorbis Files (*.ogg *.OGG)\nFLAC Files (*.flac *.FLAC)\nTM Century GoldDrive Files (*.tmc *.TMC)\nWAV Files (*.wav *.WAV)\nAll Files (*.*)"

/*
 * Image File Filter for QFileDialog
 */
#define RD_IMAGE_FILE_FILTER "Image Files (*.png *.bmp *.xbm *.xpm *.pbm *.pgm *.ppm *.jpg *.mng *.gif *.PNG *.BMP *.XBM *.XPM *.PBM *.PGM *.PPM *.JPG *.MNG *.GIF)\nAll Files (*.*)"

/*
 * Loadable Module Filter for QFileDialog
 */
#define RD_MODULE_FILE_FILTER "Rivendell Loadable Modules (*.rlm)\nAll Files (*.*)"

/*
 * Web Interface Settings
 */
#define RD_WEB_BACKGROUND_COLOR "#FFFFFF"
#define RD_WEB_LINE_COLOR1 "#E0E0E0"
#define RD_WEB_LINE_COLOR2 "#F0F0F0"

/*
 * LiveWire Values
 */
#define RD_LIVEWIRE_DEFAULT_TCP_PORT 93
#define RD_LIVEWIRE_DEFAULT_STREAM_PORT 5004
#define RD_LIVEWIRE_DEFAULT_CHANNELS 2
#define RD_LIVEWIRE_MAX_CMD_LENGTH 1024
#define RD_LIVEWIRE_GPIO_BUNDLE_SIZE 5
#define RD_LIVEWIRE_DEFAULT_LOAD RDLiveWireDestination::LoadHighZ
#define RD_LIVEWIRE_MAX_SOURCE 0x7FFF
#define RD_LIVEWIRE_GPIO_MCAST_ADDR "239.192.255.4"
#define RD_LIVEWIRE_GPIO_SEND_PORT 2055
#define RD_LIVEWIRE_GPIO_RECV_PORT 2060
#define RD_LIVEWIRE_GPIO_PULSE_WIDTH 100

/*
 * Number of Carts to Include in a 'Limited' Search
 */
#define RD_LIMITED_CART_SEARCH_QUANTITY 100

/*
 * Timeout for libcurl (secs)
 */
#define RD_CURL_TIMEOUT 1200

/*
 * Anonymous FTP Credentials
 */
#define RD_ANON_FTP_USERNAME "anonymous"
#define RD_ANON_FTP_PASSWORD "rivendell"

/*
 * Status Monitor Height
 */
#define RDMONITOR_HEIGHT 30

/*
 * RDAirPlay Colors
 */
#define RDPANEL_SETUP_FLASH_COLOR Qt::blue
#define RDPANEL_RESET_FLASH_COLOR Qt::blue
#define RDPANEL_PLAY_BACKGROUND_COLOR Qt::red
#define RDPANEL_PAUSED_BACKGROUND_COLOR Qt::cyan
#define BUTTON_STOPPED_BACKGROUND_COLOR Qt::green
#define BUTTON_PLAY_BACKGROUND_COLOR Qt::red
#define BUTTON_PLAY_TEXT_COLOR Qt::black
#define BUTTON_PAUSE_BACKGROUND_COLOR Qt::cyan
#define BUTTON_PAUSE_TEXT_COLOR Qt::black
#define BUTTON_FROM_BACKGROUND_COLOR Qt::magenta
#define BUTTON_FROM_TEXT_COLOR Qt::black
#define BUTTON_TO_BACKGROUND_COLOR Qt::yellow
#define BUTTON_TO_TEXT_COLOR Qt::black
#define BUTTON_DISABLED_BACKGROUND_COLOR Qt::darkGray
#define BUTTON_DISABLED_TEXT_COLOR Qt::white
#define BUTTON_ERROR_BACKGROUND_COLOR Qt::red
#define BUTTON_ERROR_TEXT_COLOR Qt::black

#define LOG_SCHEDULED_COLOR Qt::white
#define LOG_PLAYING_COLOR Qt::green
#define LOG_PAUSED_COLOR Qt::cyan
#define LOG_NEXT_COLOR "#CCFFCC"
#define LOG_FINISHED_COLOR Qt::gray
#define LOG_EVERGREEN_COLOR "#008000"
#define LOG_ERROR_COLOR Qt::red
#define LOG_RELATIVE_TEXT_COLOR Qt::black
#define LOG_HARDTIME_TEXT_COLOR Qt::blue

/*
 * Cart Label Box Colors
 */
#define BAR_UNCHANGED_TRANSITION_COLOR Qt::green
#define BAR_UNCHANGED_STOPPING_COLOR Qt::red
#define BAR_CHANGED_TRANSITION_COLOR Qt::yellow
#define BAR_CHANGED_STOPPING_COLOR Qt::blue
#define LABELBOX_BACKGROUND_COLOR Qt::white
#define LABELBOX_MISSING_COLOR Qt::red
#define LABELBOX_MARKER_COLOR Qt::cyan
#define LABELBOX_CHAIN_COLOR Qt::magenta
#define LABELBOX_EVERGREEN_COLOR "#008000"
#define LABELBOX_TIMESCALE_COLOR Qt::green

/*
 * RDCartSlots Settings
 */
#define RDCARTSLOTS_MAX_ROWS 16
#define RDCARTSLOTS_MAX_COLUMNS 4

/*
 * RDLogEdit Log Listing Limit Quantity
 */
#define RDLOGEDIT_LIMIT_QUAN 14

/*
 * Custom MIME Types (for Drag and Drop)
 */
#define RDMIMETYPE_CART "application/rivendell-cart"

/*
 * Number of Cut Event IDs
 */
#define RD_CUT_EVENT_ID_QUAN 16


#endif  // RD_H
